### PR TITLE
chore: Add ZKSync into the Chain type

### PR DESCRIPTION
**Pull request by syahrul12345**
## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Breaking changes

### CODE REVIEW

File: ethers-core/src/types/chain.rs [hunk 1]
```diff
@@ -460,10 +463,14 @@ impl Chain {
...
 
+            ZKSync => {
+                ("https://zksync2-mainnet-explorer.zksync.io/", "https://explorer.zksync.io/")
+            }
+
             AnvilHardhat | Dev | Morden | MoonbeamDev | FilecoinMainnet => {
                 // this is explicitly exhaustive so we don't forget to add new urls when adding a
                 // new chain
-                return None
+                return None;
```
Comment (mattsse) [inline ethers-core/src/types/chain.rs:473]:
```suggestion
                return None
```

**Review by mattsse — APPROVED** [review]
lgtm

File: ethers-core/src/types/chain.rs [hunk 2]
```diff
@@ -140,6 +140,8 @@ pub enum Chain {
     CantoTestnet = 740,
 
     Boba = 288,
+
+    ZKSync = 324,
```
Comment (DaniPopes) [inline ethers-core/src/types/chain.rs:144]: this needs to be renamed to lowercase for serde and strum, see the comments at the top of the file on how to do it

File: ethers-core/src/types/chain.rs [hunk 3]
```diff
@@ -270,9 +272,9 @@ impl Chain {
             FilecoinHyperspaceTestnet | FilecoinMainnet => 30_000,
 
             // Explicitly exhaustive. See NB above.
-            Morden | Ropsten | Rinkeby | Goerli | Kovan | XDai | Chiado | Sepolia | Moonbase |
-            MoonbeamDev | Optimism | OptimismGoerli | OptimismKovan | Poa | Sokol | Rsk |
-            EmeraldTestnet | Boba => return None,
+            Morden | Ropsten | Rinkeby | Goerli | Kovan | XDai | Chiado | Sepolia | Moonbase
+            | MoonbeamDev | Optimism | OptimismGoerli | OptimismKovan | Poa | Sokol | Rsk
+            | EmeraldTestnet | Boba | ZKSync => return None,
```
Comment (DaniPopes) [inline ethers-core/src/types/chain.rs:277]: please use the nightly toolchain to format your code
(cargo +nightly fmt)

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: ethers-core/src/types/chain.rs (hunk 3 above)
Comment (syahrul12345) [reply ethers-core/src/types/chain.rs]: I've run cargo +nigtly fmt in the project

File: ethers-core/src/types/chain.rs [hunk 4]
```diff
@@ -460,6 +463,10 @@ impl Chain {
 
             Boba => ("https://api.bobascan.com/api", "https://bobascan.com"),
 
+            ZKSync => {
```
Comment (DaniPopes) [inline ethers-core/src/types/chain.rs:466]: one last thing, this wasn't renamed

```suggestion
            Zksync => {
```

Comment (syahrul12345) [reply ethers-core/src/types/chain.rs]: resolved in another commit :D thx

**Comment by prestwich** [comment]
2 requests:
1. the S should be capitalized for rusty PascalCase - `ZkSync`

**Pull request merged**